ProcessQueue/__init__.py

```diff
@@ -95,233 +108,256 @@ def main(msg: func.QueueMessage):
             "dequeue_count": msg.dequeue_count,
         }
     )
+
+    # Grab message body from storage queues, 
+    # extract filenames for scan reports and dictionaries
     message = json.loads(message)
     body = ast.literal_eval(message["body"])
-    filename = body["blob_name"]
-    # Write File saved in Blob storage to BytesIO stream
-    with BytesIO() as input_blob:
-        # Connect to Blob Service
-        blob = BlockBlobService(connection_string=os.environ.get("STORAGE_CONN_STRING"))
-        # Download Scan Report .xlsx file as a ByteIO() stream
-        blob.get_blob_to_stream(
-            container_name="raw-reports", blob_name=filename, stream=input_blob
-        )
-        input_blob.seek(0)
-
-        # Set up API parameters:
-        api_url = os.environ.get("APP_URL") + "api/"
-        headers = {
-            "Content-type": "application/json",
-            "charset": "utf-8",
-            "Authorization": "Token {}".format(os.environ.get("AZ_FUNCTION_KEY")),
+    scan_report_blob = body["scan_report_blob"]
+    data_dictionary_blob = body["data_dictionary_blob"]
+
+    print('MESSAGE BODY >>>', body)
+
+    # If dictionary is present, download dictionary and scan report
+    if data_dictionary_blob != "None":
+        # Access data as StorageStreamerDownloader class
+        # Decode and split the stream using csv.reader()
+        container_client = blob_service_client.get_container_client("data-dictionaries")
+        blob_client = container_client.get_blob_client(data_dictionary_blob)
+        streamdownloader = blob_client.download_blob()
+        data_dictionary = list(csv.DictReader(streamdownloader.readall().decode('utf-8').splitlines()))
+
+        # Grab scan report data from blob
+        container_client = blob_service_client.get_container_client("scan-reports")
+        blob_client = container_client.get_blob_client(scan_report_blob)
+        streamdownloader = blob_client.download_blob()
+        scanreport = BytesIO(streamdownloader.readall())
+    
+    else:
+        # Else download only the scan report
+        container_client = blob_service_client.get_container_client("scan-reports")
+        blob_client = container_client.get_blob_client(scan_report_blob)
+        streamdownloader = blob_client.download_blob()
+        scanreport = BytesIO(streamdownloader.readall())

```

Comment:
These lines are identical regardless of the `if` result, so please move them outside and have only the handling of the `dat_dictionary` bits inside the `if`.